GitCommands/Git/GitModule.cs

```diff
@@ -2384,12 +2384,13 @@ public IReadOnlyList<GitStash> GetStashes(bool noLocks = false)
             string extraDiffArguments,
             CancellationToken cancellationToken)
         {
-            if (firstId.IsArtificial
-                || secondId.IsArtificial
-                || (firstBase?.IsArtificial ?? false)
-                || (secondBase?.IsArtificial ?? false))
+            // range-diff is not possible for artificial commits, use HEAD
+            string first = firstId.IsArtificial ? "HEAD" : firstId.ToString();
+            string second = secondId.IsArtificial ? "HEAD" : secondId.ToString();
+
+            if ((firstBase?.IsArtificial ?? false) || (secondBase?.IsArtificial ?? false))
             {
-                throw new ArgumentException($"Tried to get range diff for artificial commit: {firstId} and file: {secondId}");
+                throw new ArgumentException($"Cannot get range diff for artificial commit baseA:{firstBase?.ToString() ?? ""} or baseB:{secondBase?.ToString() ?? ""}.");
```

Comment:
```suggestion
                throw new ArgumentException($"Cannot get range diff for artificial commit base of A: {firstBase} or base of B: {secondBase}.");
```

Reply:
ok
(I was not expecting  $"{}" to handle null)